src/cmake_build_extension/build_extension.py

```diff
@@ -161,6 +161,10 @@ def build_extension(self, ext: CMakeExtension) -> None:
         # CMake build arguments
         build_args = ["--config", ext.cmake_build_type]
 
+        # If CMAKE_BUILD_PARALLEL_LEVEL is not set, apply the corresponding user settings.
+        if "CMAKE_BUILD_PARALLEL_LEVEL" not in os.environ:
```

Comment:
```suggestion
        if ext.cmake_parallel_jobs is not None:
```

Then, downstream projects can fill their own `setup.py` either with `cmake_parallel_jobs=X`, or checking `CMAKE_BUILD_PARALLEL_LEVEL`.